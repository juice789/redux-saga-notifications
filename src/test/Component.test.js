/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render, screen } from './helpers'
import 'jest-styled-components'
import '@testing-library/jest-dom'

import Notifications from '../Component'

it('default component, no notifications', () => {
  const { container } = render(<Notifications />)
  expect(container.firstChild).toMatchSnapshot()
  expect(screen.queryByText(/Test notification/)).not.toBeInTheDocument()
})

it('should show a notification', () => {
  const state = {
    1: {
      label: 'Test notification',
      duration: '5000',
      id: '1'
    }
  }
  const { container } = render(<Notifications />, {
    preloadedState: { notificationReducer: state }
  })
  expect(container.firstChild).toMatchSnapshot()
  expect(screen.queryByText(/Test notification/)).toBeInTheDocument()
  expect(screen.queryByText(/Yes/)).not.toBeInTheDocument()
  expect(screen.queryByText(/No/)).not.toBeInTheDocument()
})

it('should show a yes/no notification', () => {
  const state = {
    1: {
      label: 'Test notification',
      duration: '5000',
      id: '1',
      buttons: 'yesNo'
    }
  }
  const { container } = render(<Notifications />, {
    preloadedState: { notificationReducer: state }
  })
  expect(container.firstChild).toMatchSnapshot()
  expect(screen.queryByText(/Test notification/)).toBeInTheDocument()
  expect(screen.queryByText(/Yes/)).toBeInTheDocument()
  expect(screen.queryByText(/No/)).toBeInTheDocument()
})

it('should show a confirmation notification', () => {
  const state = {
    1: {
      label: 'Test notification',
      duration: '5000',
      id: '1',
      buttons: 'ok'
    }
  }
  const { container } = render(<Notifications />, {
    preloadedState: { notificationReducer: state }
  })
  expect(container.firstChild).toMatchSnapshot()
  expect(screen.queryByText(/Test notification/)).toBeInTheDocument()
  expect(screen.queryByText(/Yes/)).not.toBeInTheDocument()
  expect(screen.queryByText(/No/)).not.toBeInTheDocument()
  expect(screen.queryByText(/Ok/)).toBeInTheDocument()
})
