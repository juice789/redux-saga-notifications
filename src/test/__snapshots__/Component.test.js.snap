// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default component, no notifications 1`] = `
.c0 {
  position: fixed;
  bottom: 15px;
  width: 100%;
  height: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
}

.c1 {
  position: absolute;
  bottom: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`should show a confirmation notification 1`] = `
.c0 {
  position: fixed;
  bottom: 15px;
  width: 100%;
  height: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
}

.c1 {
  position: absolute;
  bottom: 0;
}

.c2 {
  position: relative;
  overflow: hidden;
  margin: 15px 15px 0 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
  animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 15px;
  font-size: 12px;
  color: #ededee;
  font-weight: bold;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 15px;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
  font-size: 12px;
  cursor: pointer;
  color: #ededee;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          Test notification
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Ok
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show a notification 1`] = `
.c0 {
  position: fixed;
  bottom: 15px;
  width: 100%;
  height: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
}

.c1 {
  position: absolute;
  bottom: 0;
}

.c2 {
  position: relative;
  overflow: hidden;
  margin: 15px 15px 0 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
  animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 15px;
  font-size: 12px;
  color: #ededee;
  font-weight: bold;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          Test notification
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show a yes/no notification 1`] = `
.c0 {
  position: fixed;
  bottom: 15px;
  width: 100%;
  height: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
}

.c1 {
  position: absolute;
  bottom: 0;
}

.c2 {
  position: relative;
  overflow: hidden;
  margin: 15px 15px 0 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
  animation: gtonke 0.2s ease none,iHtkXv 0.2s ease none,gDsCMX 5000ms linear none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 15px;
  font-size: 12px;
  color: #ededee;
  font-weight: bold;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 15px;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
  font-size: 12px;
  cursor: pointer;
  color: #ededee;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          Test notification
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Yes
        </div>
        <div
          class="c6"
        >
          No
        </div>
      </div>
    </div>
  </div>
</div>
`;
